src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -1161,7 +1167,19 @@ private bool IsValidDataExtensionPropertyType(ITypeSymbol type)
                 {
                     // Skip the member if either of the two conditions hold
                     // 1. Member has no accessible getters or setters (but is not marked with JsonIncludeAttribute since we need to throw a runtime exception) OR
-                    // 2. The member type is not accessible within the generated context.
+                    // 2. The member type is not accessible within the generated context OR
+                    return null;
+                }
+
+                if (memberType.IsRefLikeType)
+                {
+                    // Skip all ref-like members and emit a diagnostic
+                    // unless the property is being explicitly ignored.
```

Comment:
nit, you have longer comments in this file. No need to line-jump IMO.
```suggestion
                    // Skip all ref-like members and emit a diagnostic unless the property is being explicitly ignored.
```